sources/Algo/src/Algo.cpp

```diff
@@ -81,16 +83,53 @@ GeneratorDefinitionAlgorithm::operator()(const NodePtr& node) {
   if (node_generators.size() == 1) {
     auto model = useInfiniteReactivelimits_ ? GeneratorDefinition::ModelType::SIGNALN : GeneratorDefinition::ModelType::DIAGRAM_PQ_SIGNALN;
     const auto& gen = node_generators.front();
+
+    if (model == GeneratorDefinition::ModelType::DIAGRAM_PQ_SIGNALN && !isDiagramValid(gen)) {
+      return;
+    }
+
     generators_.emplace_back(gen.id, model, node->id, gen.points, gen.qmin, gen.qmax, gen.pmin, gen.pmax);
   } else {
     auto model =
         useInfiniteReactivelimits_ ? GeneratorDefinition::ModelType::WITH_IMPEDANCE_SIGNALN : GeneratorDefinition::ModelType::WITH_IMPEDANCE_DIAGRAM_PQ_SIGNALN;
     for (auto it = node_generators.begin(); it != node_generators.end(); ++it) {
+      if (model == GeneratorDefinition::ModelType::WITH_IMPEDANCE_DIAGRAM_PQ_SIGNALN && !isDiagramValid(*it)) {
+        continue;
+      }
       generators_.emplace_back(it->id, model, node->id, it->points, it->qmin, it->qmax, it->pmin, it->pmax);
     }
   }
 }
 
+bool
+GeneratorDefinitionAlgorithm::isDiagramValid(const inputs::Generator& generator) {
+  bool valid = true;
+
+  if (generator.points.size() == 0) {
+    return true;
```

Comment:
What does it mean to have an empty diagram?

Reply:
It means that we still have a diagram in the diagram.txt file. This diagram will be constructed from the pmin, pmax, qmin and qmax points and will give us a table(2,2)

Reply:
Could you please add a comment?

Reply:
Of course